Python test instruction is not clear

### Description

I follow exactly the documentation https://github.com/lancedb/lancedb/blob/main/python/CONTRIBUTING.md, but when I `make test`, I get the following error:
```
FAILED python/tests/test_remote_db.py::test_async_remote_db - OSError: [Errno 48] Address already in use
FAILED python/tests/test_remote_db.py::test_async_checkout - OSError: [Errno 48] Address already in use
FAILED python/tests/test_remote_db.py::test_http_error - OSError: [Errno 48] Address already in use
FAILED python/tests/test_remote_db.py::test_retry_error - OSError: [Errno 48] Address already in use
FAILED python/tests/test_remote_db.py::test_table_add_in_threadpool - OSError: [Errno 48] Address already in use
FAILED python/tests/test_remote_db.py::test_table_create_indices - OSError: [Errno 48] Address already in use
FAILED python/tests/test_remote_db.py::test_query_sync_minimal - OSError: [Errno 48] Address already in use
FAILED python/tests/test_remote_db.py::test_query_sync_empty_query - OSError: [Errno 48] Address already in use
FAILED python/tests/test_remote_db.py::test_query_sync_maximal - OSError: [Errno 48] Address already in use
FAILED python/tests/test_remote_db.py::test_query_sync_multiple_vectors - OSError: [Errno 48] Address already in use
FAILED python/tests/test_remote_db.py::test_query_sync_fts - OSError: [Errno 48] Address already in use
FAILED python/tests/test_remote_db.py::test_query_sync_hybrid - OSError: [Errno 48] Address already in use
ERROR python/tests/test_s3.py::test_s3_lifecycle - botocore.exceptions.EndpointConnectionError: Could not connect to the endpoint URL: "http://localhost:4566/lance-integtest?encoding-type=url"
ERROR python/tests/test_s3.py::test_s3_sse - botocore.exceptions.EndpointConnectionError: Could not connect to the endpoint URL: "http://localhost:4566/lance-integtest?encoding-type=url"
ERROR python/tests/test_s3.py::test_s3_dynamodb - botocore.exceptions.EndpointConnectionError: Could not connect to the endpoint URL: "http://localhost:4566/lance-integtest?encoding-type=url"
ERROR python/tests/test_s3.py::test_s3_dynamodb_sync - botocore.exceptions.EndpointConnectionError: Could not connect to the endpoint URL: "http://localhost:4566/lance-integtest?encoding-type=url"
================================================================================================================== 12 failed, 251 passed, 23 skipped, 18 deselected, 7 warnings, 4 errors in 34.87s ===================================================================================================================
make: *** [test] Error 1
```
I'm running the commands in my macbook. Please advise how to solve this, thanks!

### Link

https://github.com/lancedb/lancedb/blob/main/python/CONTRIBUTING.md

--- a/python/Makefile
+++ b/python/Makefile
@@ -29,4 +29,4 @@ doctest:	## Run documentation tests.
 
 .PHONY: test
 test:		## Run tests.
-	pytest python/tests -vv --durations=10 -m "not slow"
+	pytest python/tests -vv --durations=10 -m "not slow and not s3_test"

--- a/python/python/tests/test_embeddings.py
+++ b/python/python/tests/test_embeddings.py
@@ -341,6 +341,7 @@ def test_add_optional_vector(tmp_path):
     assert not (np.abs(tbl.to_pandas()["vector"][0]) < 1e-6).all()
 
 
+@pytest.mark.slow
 @pytest.mark.parametrize(
     "embedding_type",
     [

--- a/python/python/tests/test_remote_db.py
+++ b/python/python/tests/test_remote_db.py
@@ -32,15 +32,16 @@ def make_mock_http_handler(handler):
 @contextlib.contextmanager
 def mock_lancedb_connection(handler):
     with http.server.HTTPServer(
-        ("localhost", 8080), make_mock_http_handler(handler)
+        ("localhost", 0), make_mock_http_handler(handler)
     ) as server:
+        port = server.server_address[1]
         handle = threading.Thread(target=server.serve_forever)
         handle.start()
 
         db = lancedb.connect(
             "db://dev",
             api_key="fake",
-            host_override="http://localhost:8080",
+            host_override=f"http://localhost:{port}",
             client_config={
                 "retry_config": {"retries": 2},
                 "timeout_config": {
@@ -59,15 +60,16 @@ def mock_lancedb_connection(handler):
 @contextlib.asynccontextmanager
 async def mock_lancedb_connection_async(handler, **client_config):
     with http.server.HTTPServer(
-        ("localhost", 8080), make_mock_http_handler(handler)
+        ("localhost", 0), make_mock_http_handler(handler)
     ) as server:
+        port = server.server_address[1]
         handle = threading.Thread(target=server.serve_forever)
         handle.start()
 
         db = await lancedb.connect_async(
             "db://dev",
             api_key="fake",
-            host_override="http://localhost:8080",
+            host_override=f"http://localhost:{port}",
             client_config={
                 "retry_config": {"retries": 2},
                 "timeout_config": {

